### PR TITLE
Fix: search spore result error

### PR DESCRIPTION
**Pull request by PainterPuppets**
resolve: https://github.com/Magickbase/ckb-explorer-public-issues/issues/676

### CODE REVIEW

File: src/components/Search/utils.ts [hunk 1]
```diff
@@ -37,7 +37,7 @@ export const getURLByAggregateSearchResult = (result: AggregateSearchResult) =>
       if (attributes.udtType === 'xudt') {
         return `/xudt/${attributes.typeHash}`
       }
-      break
+      return `/sudt/${attributes.typeHash}`
```
Comment (Keith-CY) [inline src/components/Search/utils.ts:40]: In what case will `udtType` not be `sudt` but the page should be redirected to `sudt page`

Comment (PainterPuppets) [reply src/components/Search/utils.ts:40]: `spore_cell`, I didn't find the specific page, but I found that the `sudt` page can view all udt (sudt / xudt / spore etc..), so I added this as fallback

Comment (Keith-CY) [reply src/components/Search/utils.ts:40]: `Spore` was covered in `nft` page, e.g. https://explorer.nervos.org/nft-info/0x42790e8287162a26801fc14d65c1624524c25dd14ee4212c7b7150e1d4f58df5/38137753267183539873752221734122950207355577885987165145598678283704816207539

Comment (Keith-CY) [reply src/components/Search/utils.ts:40]: Please check @PainterPuppets

Comment (PainterPuppets) [reply src/components/Search/utils.ts:40]: > Please check @PainterPuppets

Got it.

Comment (PainterPuppets) [reply src/components/Search/utils.ts:40]: > Please check @PainterPuppets

https://github.com/Magickbase/ckb-explorer-public-issues/issues/676#issuecomment-2144362179

I'm a little confused as to why `spore_cell` would be an `udt` type, and then I tried to access `https://explorer.nervos.org/nft-collections/0x43ec3b508b32bc3976a9f7366facd5281e7ba801dff531ee9e4b415152376cbe ` (mentioned in issue Error Search Results) and found that the nft page doesn't show up, I'm wondering if this is a data error or something, is it normal for `spore_cell` to appear in `udt`?

Comment (Keith-CY) [reply src/components/Search/utils.ts:40]: > > Please check @PainterPuppets
> 
> [Magickbase/ckb-explorer-public-issues#676 (comment)](https://github.com/Magickbase/ckb-explorer-public-issues/issues/676#issuecomment-2144362179)
> 
> I'm a little confused as to why `spore_cell` would be an `udt` type, and then I tried to access [explorer.nervos.org/nft-collections/0x43ec3b508b32bc3976a9f7366facd5281e7ba801dff531ee9e4b415152376cbe](https://explorer.nervos.org/nft-collections/0x43ec3b508b32bc3976a9f7366facd5281e7ba801dff531ee9e4b415152376cbe) (mentioned in issue Error Search Results) and found that the nft page doesn't show up, I'm wondering if this is a data error or something, is it normal for `spore_cell` to appear in `udt`?

I got it.

`0x0632184fa0aa9363a7f3a9fbfbcf2708d953178ae75111da1e81427bc80fb1ef` is a type hash of `Spore Item Cell`, copied from https://explorer.nervos.org/transaction/0xcfc39752a33ac994ed061d12907228e4249a04a2503bd21d84c236f635d28fe6, it's a `Unicorn itme` and its page is https://explorer.nervos.org/nft-info/0xfe9ed5346d9126582241a0d3d936b47513f3ad13129557c6bb8a16576dd8b8b6/18306823470741201083262782512497888603216366300343015825462981723873045879564

where

collection id is `0xfe9ed5346d9126582241a0d3d936b47513f3ad13129557c6bb8a16576dd8b8b6`
item id is `0x28794b7e2c16ae9b6751042f102d789c31974b35693e8ac804c3c2e20f839b0c`

I think these data are missing in the result
cc @rabbitz

Comment (rabbitz) [reply src/components/Search/utils.ts:40]: PR: https://github.com/nervosnetwork/ckb-explorer/pull/1958

1. staging search spore item type_hash `0xe158964434ec4984aa6f6ea8676de7c5b685032ba52514cccbb837c62da8a418`, the result is 

```ruby
{
    "data": [
        {
            "id": "205338",
            "type": "token_item",
            "attributes": {
                "name": null,
                "icon_url": null,
                "metadata_url": null,
                "status": "normal",
                "token_id": 4217419134417764539785820948464239161488642572777684692133525740984935036370,
                "token_collection": {
                    "standard": "spore",
                    "name": "Cluster name",
                    "description": "Description of the cluster",
                    "icon_url": null,
                    "symbol": null,
                    "sn": "0xbf7a42821163a5940d6cf66042cce7c9d6e6de86c6ea8186a2e642b6532b77b6"
                }
            }
        }
    ]
}
```

2. search token type hash `0xbf7a42821163a5940d6cf66042cce7c9d6e6de86c6ea8186a2e642b6532b77b6` the result is

```ruby
{
    "data": [
        [
            {
                "id": "18965",
                "type": "token_collection",
                "attributes": {
                    "standard": "spore",
                    "name": "Cluster name",
                    "description": "Description of the cluster",
                    "icon_url": null,
                    "symbol": null,
                    "sn": "0xbf7a42821163a5940d6cf66042cce7c9d6e6de86c6ea8186a2e642b6532b77b6"
                }
            }
        ]
    ]
}
```

**Comment by PainterPuppets** [comment]
image

**Review by Keith-CY — APPROVED** [review]

**Comment by FrederLu** [comment]
0x01e71a9d61a7423164e420c8e9e12bd23b8008bfbff41b2864142ccd809ed98f
image
0x641860502c55ef0f6dde30d83353c336fc95ec572cdafa8f296974286f46a2b7
image
image
Search results can be obtained using script hash, but the redirect page is 404.

**Comment by Keith-CY** [comment]
> 0x01e71a9d61a7423164e420c8e9e12bd23b8008bfbff41b2864142ccd809ed98f image 0x641860502c55ef0f6dde30d83353c336fc95ec572cdafa8f296974286f46a2b7 image image Search results can be obtained using script hash, but the redirect page is 404.

It was led to https://ckb-explorer-frontend-in-magickbase-repo-git-ea5e0c-magickbase.vercel.app/nft-collections/0xcf9e0cdbd169550492b29d3d1181d27048ab80126b797840965d2864607a892d/92096543841238977728566825683252965277957597261053391592605747421671503922149
but the correct url should be
https://ckb-explorer-frontend-in-magickbase-repo-git-ea5e0c-magickbase.vercel.app/nft-info/0xcf9e0cdbd169550492b29d3d1181d27048ab80126b797840965d2864607a892d/92096543841238977728566825683252965277957597261053391592605747421671503922149

<del>nft-collections</del>nft-info

File: src/components/Search/utils.ts [hunk 2]
```diff
@@ -42,6 +42,9 @@ export const getURLByAggregateSearchResult = (result: AggregateSearchResult) =>
     case SearchResultType.TokenCollection:
       return `/nft-collections/${attributes.sn}`
 
+    case SearchResultType.TokenItem:
+      return `/nft-collections/${attributes.tokenCollection.sn}/${attributes.tokenId}`
```
Comment (Keith-CY) [inline src/components/Search/utils.ts:46]:
```suggestion
      return `/nft-info/${attributes.tokenCollection.sn}/${attributes.tokenId}`
```

**Review by FrederLu — APPROVED** [review]

**Pull request merged**